packages/expo-file-system/ios/EXFileSystem/EXSessionTasks/EXSessionResumableDownloadTaskDelegate.h

```diff
@@ -0,0 +1,20 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+#import <EXFileSystem/EXSessionDownloadTaskDelegate.h>
+#import <EXFileSystem/EXResumablesManager.h>
+
+typedef void (^EXDownloadDelegateOnWriteCallback)(NSURLSessionDownloadTask *task, int64_t bytesWritten, int64_t totalBytesWritten, int64_t totalBytesExpectedToWrite);
+
+@interface EXSessionResumableDownloadTaskDelegate : EXSessionDownloadTaskDelegate
+
+- (instancetype)initWithResolve:(UMPromiseResolveBlock)resolve
+                         reject:(UMPromiseRejectBlock)reject
+                       localUrl:(NSURL *)localUrl
+             shouldCalculateMd5:(BOOL)shouldCalculateMd5
+                onWriteCallback:(EXDownloadDelegateOnWriteCallback)onWriteCallback
+               resumableManager:(EXResumablesManager *)manager
+                           uuid:(NSString *)uuid;
+
+- (void)invalid;
```

Comment:
1. It's more common to use a verb in methods names, so it could be `invalidate`.
2. As per comment in task dispatcher we should be able to remove this extra function in favor of invalidating the delegate ourselves (in the implemented callback).